### PR TITLE
[Feature] Adds special item rewards to fixed classic/challenge battles

### PR DESCRIPTION
**Pull request by ben-lear**
## What are the changes the user will see?
Custom items after special battles in Classic/Challenge mode

## Why am I making these changes?
To give custom rewards for special battles

## What are the changes from a developer perspective?
- 25 Rival 2: 1 Ultra 2 Great Items
- 55 Rival 3: 2 Ultra 2 Great Items
- 95 Rival 4:  4 Ultra Items
- 115 Evil Boss 1: 3 Ultra 2 Rogue Items
- 145 Rival 5: 2 Ultra 3 Rogue Items
- 165 Evil Boss 2: Lock Capsule +4 Ultra 2 Rogue Items
- 195 Rival 5: 2 Rogue 2 Ultra 2 Great

none of these items can upgrade tiers from luck on the initial roll

### Screenshots/Videos

https://github.com/user-attachments/assets/beb3d510-5645-4e2d-8a9a-249606a61218

https://github.com/user-attachments/assets/43f96c31-7ce3-40e3-8f67-6c6ea9f09d6e

## How to test the changes?
locally, override floor to one of the specified floors and check the rewards are correct

## Checklist
- [ ] **I'm using `beta` as my base branch**
- [ ] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
- [ ] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by damocleas — CHANGES_REQUESTED** [review]
Please remove Lock Capsule from classic in pokerogue / src / modifier / modifier-type.ts

**Review by damocleas — APPROVED** [review]

File: src/modifier/modifier-type.ts [hunk 1]
```diff
@@ -1762,7 +1762,7 @@ const modifierPool: ModifierPool = {
     new WeightedModifierType(modifierTypes.FOCUS_BAND, 5),
     new WeightedModifierType(modifierTypes.KINGS_ROCK, 3),
     new WeightedModifierType(modifierTypes.LOCK_CAPSULE, skipInLastClassicWaveOrDefault(3)),
```
Comment (ben-lear) [inline src/modifier/modifier-type.ts]:
```suggestion
    new WeightedModifierType(modifierTypes.LOCK_CAPSULE, (party: Pokemon[]) => party[0].scene.gameMode.isClassic ? 0 : 3)),
```

File: src/modifier/modifier-type.ts [hunk 2]
```diff
@@ -1762,7 +1762,7 @@ const modifierPool: ModifierPool = {
...
-    new WeightedModifierType(modifierTypes.SUPER_EXP_CHARM, skipInLastClassicWaveOrDefault(8)),
+    new WeightedModifierType(modifierTypes.SUPER_EXP_CHARM, (party: Pokemon[]) => party[0].scene.gameMode.isClassic ? 0 : 8),
```
Comment (ben-lear) [inline src/modifier/modifier-type.ts]:
```suggestion
    new WeightedModifierType(modifierTypes.SUPER_EXP_CHARM, skipInLastClassicWaveOrDefault(8)),
```

File: src/phases/victory-phase.ts [hunk 3]
```diff
@@ -42,8 +42,12 @@ export class VictoryPhase extends PokemonPhase {
       }
       if (this.scene.gameMode.isEndless || !this.scene.gameMode.isWaveFinal(this.scene.currentBattle.waveIndex)) {
         this.scene.pushPhase(new EggLapsePhase(this.scene));
+        if (this.scene.gameMode.isClassic && this.scene.currentBattle.waveIndex === 165) {
```
Comment (flx-sta) [inline src/phases/victory-phase.ts:45]: NIT: 
`165` is a magic number.
Should possibly be moved to a constant

File: src/phases/victory-phase.ts [hunk 4]
```diff
@@ -76,4 +80,18 @@ export class VictoryPhase extends PokemonPhase {
 
     this.end();
   }
+
+  /**
+   * If this wave is a fixed battle with special custom modifier rewards,
+   * will pass those settings to the upcoming `SelectModifierPhase`.
```
Comment (flx-sta) [inline src/phases/victory-phase.ts:86]: NIT: 
```suggestion
   * will pass those settings to the upcoming {@linkcode SelectModifierPhase}.
```

**Review by flx-sta — DISMISSED** [review]
only NITs

Nothing crucial

File: src/modifier/modifier-type.ts [hunk 5]
```diff
@@ -1761,7 +1761,7 @@ const modifierPool: ModifierPool = {
     new WeightedModifierType(modifierTypes.ABILITY_CHARM, skipInClassicAfterWave(189, 6)),
     new WeightedModifierType(modifierTypes.FOCUS_BAND, 5),
     new WeightedModifierType(modifierTypes.KINGS_ROCK, 3),
-    new WeightedModifierType(modifierTypes.LOCK_CAPSULE, skipInLastClassicWaveOrDefault(3)),
+    new WeightedModifierType(modifierTypes.LOCK_CAPSULE, (party: Pokemon[]) => party[0].scene.gameMode.isClassic ? 0 : 3),
```
Comment (innerthunder) [inline src/modifier/modifier-type.ts:1764]: Is this meant to get rid of Lock Capsule in Classic altogether? 🤔

Comment (damocleas) [reply src/modifier/modifier-type.ts:1764]: Yes, please check the other files. It is given on a set wave now.

**Review by innerthunder — APPROVED** [review]

**Review by damocleas — APPROVED** [review]

**Review by innerthunder — APPROVED** [review]

**Pull request merged**